Sorting/intersection2.js

```diff
@@ -0,0 +1,32 @@
+//definition of intersection, built off a helper function. repetitions in subArrays won't break it.
+
+//helper function describes how to intersect two arrays.
+//checks whether every element in arr1, is in arr2 + not already in intersection array (this avoids repetitions)
+//if element satisfies this condition, it pushes the element into the output array
+function intersectTwoArrays(arr1, arr2) {
+    var result = [];
+    arr1.forEach(function(element) {
+        if (arr2.includes(element) && !result.includes(element)) {
+            result.push(element);
+        }
+    })
+    return result;
+}
+
+
+//if arrOfArrs.length < 2, return nothing or the only subArray.
+//else, this function establishes the base as the intersection of the first two arrays.
+//then it cycles through all subsequent subArrays, and updates the value of intersection as the intersection between the current intersection and the next subArray. 
+function intersectArrayOfArrays(arrOfArrs) {
+     if (arrOfArrs.length === 0) {
+       return;
+     }
+     if (arrOfArrs.length === 1) {
+       return arrOfArrs[0];
+     }
+     var intersection = intersectTwoArrays(arrOfArrs[0], arrOfArrs[1]);
```

Comment:
Very, very cool! This is a general pattern called `reduce` (also known as `fold` in some languages). It appears very often in many different problems.

The idea of `reduce` is that you take a function of two arguments (in this case `intersectTwoArrays`) and use it to `reduce` a list of values into a single value. 

(Perhaps you've heard of [MapReduce](https://en.wikipedia.org/wiki/MapReduce)? The "reduce" in the name refers to this pattern.)

Javascript even has a [function on arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) that supports it.  Here is a post with [javascript reduce examples](https://medium.freecodecamp.org/reduce-f47a7da511a9) that can be implemented with this pattern.

**Bonus**: see how you can simplify the implementation of `intersectArrayOfArrays` using javascript's `reduce` function.

Reply:
function intersectArrayOfArrays(arrOfArrs) {
  return arrOfArrs.reduce(intersectTwoArrays, arrOfArrs[0]);
}

Reply:
putting four spacees before a line
    will format your comments like this

or you can use two backtics: \`like this\`     `like this`

😉 😛 😳

Reply:
Good point, @fenimore! There is no need to pass in the first argument. 

Here is the signature of javascript's reduce:

```arr.reduce(callback[, initialValue])```

The brackets in `[, initialValue]` mean that the second argument, the initial value for the reduce, is _optional_. This means that while @rictorlome 's solution is correct, it can be simplified further to: 

```javascript
function intersectArrayOfArrays(arrOfArrs) {
   return arrOfArrs.reduce(intersectTwoArrays);
}
```

I believe the behavior is as follows. Suppose `arrOfArrs` is `[w, x, y, z]`, then:

If there is an `initialValue` of `b` then the reduce would look like:
```javascript
arrOfArrs.reduce(intersectTwoArrays, b)   is equivalent to intersectTwoArrays(intersectTwoArrays(intersectTwoArrays(intersectTwoArrays(b, w), x), y), z)
```

However, without an initial value, the first two elements are combined directly to kick off the reduce, resulting in one fewer calls to the `intersectTwoArrays` function:
```javascript
arrOfArrs.reduce(intersectTwoArrays)   is equivalent to intersectTwoArrays(intersectTwoArrays(intersectTwoArrays(w, x), y), z)
```

---

The code using the accumulator looks good. I wonder how your `function` compares to @rictorlome 's `intersectTwoArrays`.

---

Nice tips about the markdown formatting. While writing this comment, I see a link under the box with a guide for how to [format the markdown](https://guides.github.com/features/mastering-markdown/).

I like that you can enable syntax highlighting by doing this above the three backticks:
```
\`\`\`javascript
if (isAwesome){
  return true
}
\`\`\`
```

Reply:
Oh I think I misinterpreted the problem! 😳

I thought it was deduplicate... But it's where they _intersect_